engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/AdditionalDictComponentConfigsExtractor.scala

```diff
@@ -0,0 +1,28 @@
+package pl.touk.nussknacker.engine.management.periodic
+
+import pl.touk.nussknacker.engine.api.component.{ComponentAdditionalConfig, DesignerWideComponentId}
+import pl.touk.nussknacker.engine.api.parameter.ValueInputWithDictEditor
+
+object AdditionalDictComponentConfigsExtractor {
+
+  // This function filters additional configs provided by AdditionalUIConfigProvider
+  // to include only component and parameter configs with Dictionary editors.
+  // This is done to reduce data sent to Flink as only configs regarding this editor are required to be known during execution.
+  def getAdditionalConfigsWithDictParametersEditors(
+      additionalComponentConfigs: Map[DesignerWideComponentId, ComponentAdditionalConfig]
+  ): Map[DesignerWideComponentId, ComponentAdditionalConfig] = {
+    additionalComponentConfigs
```

Comment:
but in this place, we can have a private method that describes what we want to leave. In the future, someone would need to create another one. But the public interface won't change.